src/components/grid/grid-item/grid-item.d.ts

```diff
@@ -0,0 +1,58 @@
+import * as React from "react";
+
+interface GridPaddingProps {
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding */
+  p?: string | number;
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding-left */
+  pl?: string | number;
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding-right */
+  pr?: string | number;
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding-top */
+  pt?: string | number;
+  /** Any valid CSS value or a number to be multiplied by base spacing unit (8). Overrides default padding-bottom */
+  pb?: string | number;
+}
+
+interface ResponsiveSettingsShape extends GridPaddingProps {
```

Comment:
This could just extend `PaddingProps` rather than `GridPaddingProps` and then you could remove `px` and `py` below as well

Reply:
as with my previous comment - Grid does not implement `styled-system` yet